comp/core/workloadmeta/collectors/internal/process/process_collector.go

```
@@ -0,0 +1,237 @@
+		store:                  store,
+		catalog:                catalog,
+		clock:                  clock,
+		processProbe:           processProbe,
+		processEventsCh:        processEventsCh,
+		lastCollectedProcesses: lastCollectedProcesses,
+	}
+}
+
+// NewProcessCollectorProvider returns a new process collector provider and an error.
+// Currently, this is only used on Linux when language detection and run in core agent are enabled.
+func NewProcessCollectorProvider() (workloadmeta.CollectorProvider, error) {
+	collector := newProcessCollector(collectorID, nil, workloadmeta.NodeAgent, clock.New(), procutil.NewProcessProbe(), make(chan *Event), make(map[int32]*procutil.Process))
+	return workloadmeta.CollectorProvider{
+		Collector: &collector,
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewProcessCollectorProvider)
+}
+
+// isEnabled returns a boolean indicating if the process collector is enabled and what collection interval to use if it is.
+func (c *collector) isEnabled() bool {
+	// TODO: implement the logic to check if the process collector is enabled based on dependent configs (process collection, language detection, service discovery)
+	// hardcoded to false until the new collector has all functionality/consolidation completed (service discovery, language collection, etc)
+	return false
+}
+
+// collectionIntervalConfig returns the configured collection interval
+func (c *collector) collectionIntervalConfig() time.Duration {
+	// TODO: read configured collection interval once implemented
+	return time.Second * 10
+}
+
+// Start starts the collector. The collector should run until the context
+// is done. It also gets a reference to the store that started it so it
+// can use Notify, or get access to other entities in the store.
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	enabled := c.isEnabled()
+
+	if enabled {
```

Nit optional: could we throw an error here if the collector is started when it's not meant to be (similar to the docker, kubelet, etc. collectors).

`if ( ... ) { return errors.NewDisabled(componentName, "Agent is not running Linux Process collector") }`

---

Added a basic check + log if process collection is disabled. 

As for next steps, there are floating questions about if this collection should enabled by default because of the service discovery consolidation + language detection, so we will likely leave this as is until we are all synced.